### PR TITLE
gh-94309: Update "What's New for Python 3.12"

### PR DESCRIPTION
**Pull request by MonadChains**
#94309

* Issue: gh-94309

### CODE REVIEW

File: Doc/whatsnew/3.12.rst [hunk 1]
```diff
@@ -187,6 +187,8 @@ although there is currently no date scheduled for their removal.
   immediately followed by one of keywords :keyword:`and`, :keyword:`else`,
   :keyword:`for`, :keyword:`if`, :keyword:`in`, :keyword:`is` and :keyword:`or`.
   In a future release it will be changed to a syntax error. (:gh:`87999`)
+* :class:`typing.Hashable` and :class:`typing.Sized` aliases for :class:`collections.abc.Hashable`
+  and :class:`collections.abc.Sized`. (:gh:`94309`)
```
Comment (hugovk) [inline Doc/whatsnew/3.12.rst]: Let's group it with the other `typing` deprecation. Please could you move it up, something like this? Thanks!

```rst
The following APIs were deprecated in earlier Python versions and will be removed,
although there is currently no date scheduled for their removal.

* :class:`typing.Hashable` and :class:`typing.Sized` aliases for :class:`collections.abc.Hashable`
  and :class:`collections.abc.Sized`. (:gh:`94309`)

* :class:`typing.Text` (:gh:`92332`)

* Currently Python accepts numeric literals immediately followed by keywords,
```

**Review by hugovk — APPROVED** [review]
Thank you!

File: Doc/whatsnew/3.12.rst [hunk 2]
```diff
@@ -189,6 +192,7 @@ although there is currently no date scheduled for their removal.
   In a future release it will be changed to a syntax error. (:gh:`87999`)
 
 
+
```
Comment (Fidget-Spinner) [inline Doc/whatsnew/3.12.rst]:
```suggestion

```

File: Doc/whatsnew/3.12.rst [hunk 3]
```diff
@@ -188,7 +191,6 @@ although there is currently no date scheduled for their removal.
   :keyword:`for`, :keyword:`if`, :keyword:`in`, :keyword:`is` and :keyword:`or`.
   In a future release it will be changed to a syntax error. (:gh:`87999`)
 
-
```
Comment (Fidget-Spinner) [inline Doc/whatsnew/3.12.rst]: I'm sorry. I tried removing the extra blank line but removed one line extra. Could you please add back a newline?

**Review by Fidget-Spinner — APPROVED** [review]

**Comment by JelleZijlstra** [comment]
I'm not sure I'm on board with "will be removed". Deprecated and discouraged, sure, but I don't see much reason to ever actually remove them.

**Comment by hugovk** [comment]
@JelleZijlstra Would you like another heading+intro for these two `typing` bullets?

**Comment by MonadChains** [comment]
@JelleZijlstra do you plan to remove the aliases deprecate with PEP 585 in the future? If so, I think that also these aliases should be removed in the future for consistency.

**Comment by JelleZijlstra** [comment]
I don't want the PEP 585 deprecations removed either. https://discuss.python.org/t/concern-about-pep-585-removals/15901/3?u=jelle

**Comment by Fidget-Spinner** [comment]
Woops yeah I forgot that these have no planned removal date.

**Review by Fidget-Spinner — APPROVED** [review]

**Comment by Fidget-Spinner** [comment]
@MonadChains since you asked earlier about what to contribute to. Apart from easy issues on the issues page, you can try focus on areas that you're interested in or areas you think the most help.

[Reviewing PR](https://devguide.python.org/getting-started/pull-request-lifecycle/#how-to-review-a-pull-request) is also good. However, it may require some expertise in the subject area first. So I recommend working on documentation and eventually learning how things in your favorite area work.

**Pull request merged**